Src/VsSpecific/VsSpecific/CSharpScriptRunner.cs

```diff
@@ -0,0 +1,109 @@
+﻿#if VS_SPECIFIC_2017
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.CodeAnalysis.Scripting;
+using System.IO;
+using Microsoft.CodeAnalysis.CSharp.Scripting;
+using System.Reflection;
+using EnvDTE;
+
+namespace Vim.VisualStudio.Specific
+{
+    [Export(typeof(IVimBufferCreationListener))]
+    internal sealed class CSharpScriptRunner : IVimBufferCreationListener, IDisposable

```

Comment:
I think the structure for running scripts should be the following:

1. Define a method on the host for running scripts `IVimHost.RunCSharpScript` 
1. Define a MEF interface for running scripts: `ICSharpScriptExecutor`
1. Export that interface here instead of `IVimBufferCreationListener` 
1. Optionally import `ICSharpScriptExecutor` inside `IVimHost`.

That pattern is much closer to how other features like this have been done. I feel like it's the best approach here.